### PR TITLE
[JEP-305] Incrementals

### PR DESCRIPTION
**Pull request by jglick**
Downstream of https://github.com/jenkinsci/pom/pull/20. [Example of usage from `junit`](https://github.com/jenkinsci/junit-plugin/compare/master...jglick:incrementals)

### CODE REVIEW

**Comment by jglick** [comment]
Currently consider this on hold only pending [JENKINS-50695](https://issues.jenkins-ci.org/browse/JENKINS-50695) (sanity check of IDEA workflow). @jenkinsci/code-reviewers

**Comment by jglick** [comment]
On second thought, I am not considering this on hold at all. If the metadata causes serious problems for some workflow, tool, process, etc., then we can just stop using it unless and until the problems are solved; there is no compatibility risk that I can see.

File: war/pom.xml [hunk 1]
```diff
@@ -429,6 +429,7 @@ THE SOFTWARE.
         <artifactId>maven-license-plugin</artifactId>
         <!-- version specified in grandparent pom -->
         <configuration>
+          <!-- TODO does not correctly set resolved ${project.version} for jenkins-* artifacts in Incrementals mode -->
```
Comment (batmat) [inline war/pom.xml]: Care to clarify? I opened jenkins-war at https://repo.jenkins-ci.org/incrementals/org/jenkins-ci/main/jenkins-war/2.119-rc15706.1e79797768ee/jenkins-war-2.119-rc15706.1e79797768ee.war and FWICT the value is correct:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<l:dependencies xmlns:l='licenses' groupId='org.jenkins-ci.main' artifactId='jenkins-war' version='2.119-rc15706.1e79797768ee'><l:dependency name='Jenkins war' groupId='org.jenkins-ci.main' artifactId='jenkins-war' version='2.119-rc15706.1e79797768ee' url='http://jenkins-ci.org/jenkins-parent/jenkins-war/'><l:description>Creates a war file. Also includes additional static web resources, such as images, CSS, JavaScript, and some HTML files.</l:description><l:license name='The MIT license' 
...
```

File: Jenkinsfile [hunk 2]
```diff
@@ -115,15 +122,3 @@ void withMavenEnv(List envVars = [], def body) {
         body.call()
     }
 }
-
-void renameFiles(def files, String prefix) {
```
Comment (batmat) [inline Jenkinsfile]: Why did you remove this? IIUC it was added to work around JUnit result bad aggregation between parallel branches.

**Review by batmat — APPROVED** [review]
LGTM, and should be easily adjustable, or revertable, if proving to be causing issues.

File: Jenkinsfile (hunk 2 above)
Comment (jglick) [reply Jenkinsfile]: Only because we had been publishing artifacts from both Linux and Windows branches. There was no point in that; the Windows build is there to run tests on Windows, but the artifacts should be (nearly) identical. Now the build is publishing artifacts only from the Linux branch.

**Comment by jglick** [comment]
Test failure is [also in `master`](https://ci.jenkins.io/job/Core/job/jenkins/job/master/886/testReport/hudson.util/ProcessTreeKillerTest/Linux___Linux_Publishing___considersKillingVetos/). Environmental change?

```
java.lang.AssertionError: Process should have been excluded from the killing
	at org.junit.Assert.fail(Assert.java:88)
	at hudson.util.ProcessTreeKillerTest.considersKillingVetos(ProcessTreeKillerTest.java:137)
```

**Review by oleg-nenashev — APPROVED** [review]
🐝 as long as it does not impact the standard release process

**Comment by jglick** [comment]
> as long as it does not impact the standard release process

As far as I know it should not. As noted in the [guide](https://github.com/jenkinsci/pom/blob/master/incrementals.md#running-maven-releases), metadata will need to be fixed up again after a weekly release.
